lib/libmantalite.js

```diff
@@ -19,34 +19,86 @@ module.exports.normalizeMantaPath = normalizeMantaPath;
 module.exports.authorize = authorize;
 
 /**
- * Given an resource like `/mark/stor/my%resume.pdf`, this API will convert to
- * `/mark/stor/my resume.pdf`; if you optionally pass in an `account` object as
+ * Given a resource like `/mark/buckets/my%20bucket`, this API will convert to
+ * `/mark/buckets/my bucket`; if you optionally pass in an `account` object as
  * returned from mahi.js (in this lib), the path will be converted to
- * `/d3d38d1a-e26c-11e2-9a8a-475696be42f1/stor/my resume.pdf`, which is the
+ * `/d3d38d1a-e26c-11e2-9a8a-475696be42f1/buckets/my bucket`, which is the
  * correct normal form everwhere in the system.
  *
+ * This function also acts as a form of validation for URL form.
+ *
  * Note that because `decodeURIComponent` may throw, this function takes a
  * callback, even though it is not blocking.
  */
 function normalizeMantaPath(opts, cb) {
     assert.object(opts, 'options');
+    assert(opts.log, 'options.log');
     assert.string(opts.path, 'options.path');
     assert.optionalObject(opts.account, 'options.account');
+    assert.func(cb, 'cb');
 
-    var err = null;
+    var log = opts.log;
     var key;
 
     try {
-        key = opts.path.split('/').map(decodeURIComponent).join('/');
+        /*
+         * An example URL is:
+         *
+         * /account/buckets/bucket_name/objects/object_name
+         *
+         * so parts will be:
+         * [
+         *   ""
+         *   "account",
+         *   "buckets",
+         *   "bucket_name",
+         *   "objects",
+         *   "object_name"
+         * ]
+         *
+         * This gets validated belowe
+         */
+        var parts = opts.path.split('/');
+
+        assert.equal(parts[0], '', 'blank first element');
+
+        // account
+        if (parts[1]) {
+            parts[1] = decodeURIComponent(parts[1]);
+        }
+
+        if (parts[2]) {
+            assert.equal(parts[2], 'buckets');
```

Comment:
I worry about hardcoding this in a handler that's used for every request. I think could be deferred to the route matching done by restify.